packages/miniflare/test/plugins/kv/assets.spec.ts

```diff
@@ -0,0 +1,220 @@
+import assert from "assert";
+import fs from "fs/promises";
+import path from "path";
+import anyTest, { Macro, TestFn } from "ava";
+import esbuild from "esbuild";
+import { Miniflare } from "miniflare";
+import { useTmp } from "../../test-shared";
+
+const FIXTURES_PATH = path.resolve(
+	__dirname,
+	"..",
+	"..",
+	"..",
+	"..",
+	"test",
+	"fixtures",
+	"assets"
+);
+const MODULES_ENTRY_PATH = path.join(FIXTURES_PATH, "modules.ts");
+
+interface Context {
+	serviceWorkerPath: string;
+	modulesPath: string;
+}
+
+const test = anyTest as TestFn<Context>;
+
+test.before(async (t) => {
+	// Build fixtures
+	const tmp = await useTmp(t);
+	await esbuild.build({
+		entryPoints: [MODULES_ENTRY_PATH],
+		format: "esm",
+		external: ["__STATIC_CONTENT_MANIFEST"],
+		bundle: true,
+		sourcemap: true,
+		outdir: tmp,
+	});
+	t.context.modulesPath = path.join(tmp, "modules.js");
+});
+
+type Route = keyof typeof routeContents;
+const routeContents = {
+	"/": "<p>Index</p>",
+	"/a.txt": "a",
+	"/b/b.txt": "b",
+};
+
+const getMacro: Macro<[{}, Set<Route>], Context> = {
+	async exec(t, options, expectedRoutes) {
+		const tmp = await useTmp(t);
+		for (const [route, contents] of Object.entries(routeContents)) {
+			const routePath = path.join(tmp, route === "/" ? "index.html" : route);
+			await fs.mkdir(path.dirname(routePath), { recursive: true });
+			await fs.writeFile(routePath, contents, "utf8");
+		}
+
+		const mf = new Miniflare({
+			...options,
+			scriptPath: t.context.modulesPath,
+			modules: true,
+			assetsPath: tmp,
+		});
+		t.teardown(() => mf.dispose());
+
+		for (const [route, expectedContents] of Object.entries(routeContents)) {
+			const res = await mf.dispatchFetch(`http://localhost:8787${route}`);
+			const expected = expectedRoutes.has(route as Route);
```

Comment:
I can't see any tests where this would be true?

Reply:
I might be misunderstanding what you're asking, but isn't [this test](https://github.com/cloudflare/workers-sdk/pull/6403/files#diff-9721850ebe796f312beb43aaba5cc547fd069855bb449e9db21dea1ce82e920dR76-R81) confirming that?